src/docs/sphinx/dev_guide/weak_form.rst

```diff
@@ -0,0 +1,266 @@
+.. _header-n2:
+
+=========
+Weak Form
+=========
+
+``WeakForm`` is a class that is used specify and evaluate
+finite-element-type calculations. For example, the weighted residual for
+a solid mechanics simulation may look something like:
+
+.. math::
+
+   r(u) := 
+   \underbrace{\int_\Omega \nabla\psi : \sigma(\nabla u) \; \text{d}v}_{\text{stress response}}
+   \;+\;
+   \underbrace{\int_\Omega \psi \cdot b(\mathbf{x}) \; \text{d}v}_{\text{body forces}} 
+   \;+\;
+   \underbrace{\int_{\partial\Omega} \psi \cdot \mathbf{t}(\mathbf{x}) \; \text{d}a}_{\text{surface loads}},
+
+where :math:`\psi` are the test basis functions. To describe this
+residual using ``WeakForm``, we first create the object itself, providing a
+template parameter that expresses the test and trial spaces (i.e. the
+"inputs" and "outputs" of the residual function, :math:`r`). In this
+case, solid mechanics uses H1 test and trial spaces, so we write:
```

Comment:
This is the first reference to H1.  Should it be defined here?

Reply:
```suggestion
case, solid mechanics uses nodal displacements and residuals (i.e. H1 test and trial spaces), so we write:
```